org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/handler/StackTraceRequestHandler.java

```diff
@@ -107,8 +107,13 @@ public void handle(Command command, Arguments arguments, Response response, IDeb
             sourceName = enclosingType.substring(enclosingType.lastIndexOf('.') + 1) + ".java";
             relativeSourcePath = enclosingType.replace('.', '/') + ".java";
         }
-        ISourceLookUpProvider sourceProvider = context.getProvider(ISourceLookUpProvider.class);
-        String uri = sourceProvider.getSourceFileURI(fullyQualifiedName, relativeSourcePath);
+
+        final String finalRelativeSourcePath = relativeSourcePath;
+        // use a lru cache for better performance
+        String uri = context.getSourceLookupCache().computeIfAbsent(fullyQualifiedName, key ->

```

Comment:
>computeIfAbsent [](start = 52, length = 15)

question, how to check whether the entry is  dirty?

Reply:
no need, since fullyQualifiedName=> uri will never be changed during a debug session.